src/util/base.cc

```
@@ -0,0 +1,207 @@
+      "outfile,o", bpo::value<std::string>(&m_OptOutfile), "output file");
+}
+
+void BaseCommand::PrintUsage(const std::string& name) const
+{
+  LOG(info) << "Syntax: " << name << " encode <inFile> <outfile>";
+  LOG(info) << "or    : " << name << " decode <inFile> <outfile>";
+}
+
+template <typename ToBase, typename ToByte, typename SizeConst>
+bool process(
+    ToBase ByteStreamToBase,
+    ToByte BaseToByteStream,
+    SizeConst Size,
+    bool encode,
+    kovri::core::InputFileStream* input,
+    kovri::core::OutputFileStream* output)
+{
+  std::uint8_t in_buffer[Size + 1];
+  std::uint8_t out_buffer[Size * 2 + 1];
+  std::size_t ret(0);
+  memset(in_buffer, 0, sizeof(in_buffer));
+  memset(out_buffer, 0, sizeof(out_buffer));
+  do
+    {
+      if (!input->Read(in_buffer, Size))
+        return false;
+
+      if (encode)
+        ret = ByteStreamToBase(
+            in_buffer,
+            input->Count(),
+            reinterpret_cast<char*>(out_buffer),
+            Size * 2);
+      else
+        ret = BaseToByteStream(
+            reinterpret_cast<char*>(in_buffer),
+            input->Count(),
+            out_buffer,
+            Size * 2);
+
+      if (input->Count() && !ret)
+        {
+          LOG(error) << "Error : Stream processing failed !";
+          return false;
+        }
+
+      if (!output->Write(out_buffer, ret))
+        return false;
+    } while (input->Good());
+  return true;
+}
+
+bool BaseCommand::Impl(
+    const std::string& cmd_name,
+    int argc,
+    const char* argv[])
+{
+  bpo::variables_map vm;
+  bpo::positional_options_description pos;
+  pos.add("type", 1).add("infile", 1).add("outfile", 1);
+  try
+    {
+      bpo::parsed_options parsed = bpo::command_line_parser(argc, argv)
+                                       .options(m_Desc)
+                                       .positional(pos)
+                                       .run();
+      bpo::store(parsed, vm);
+      bpo::notify(vm);
+    }
+  catch (...)
```

Thx. This was a change requested by anonimal. I think the idea is not to catch all exceptions, but  to delegate the management to Exception::Dispatch and have a unifed way of dealing with it.